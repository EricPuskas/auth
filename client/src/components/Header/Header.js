import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import "./Header.scss";

const Header = ({ auth: { user } }) => {
  const { firstName, lastName, isAuthenticated } = user;
  return (
    <div className="Header" data-test="header">
      {isAuthenticated ? (
        <h1 data-test="header-auth"> Welcome {`${firstName} ${lastName}`}!</h1>
      ) : (
        <h1 data-test="header-no-auth"> Log in to your account </h1>
      )}
    </div>
  );
};

const mapStateToProps = state => ({
  auth: state.auth
});

Header.propTypes = {
  auth: PropTypes.object
};

export default connect(
  mapStateToProps,
  null
)(Header);
