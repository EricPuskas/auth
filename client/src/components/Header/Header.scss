.Header {
  margin-top: 2rem;
  text-align: center;
  h1 {
    color: #fff;
    font-size: 1.7rem;
    text-transform: uppercase;
  }
}
