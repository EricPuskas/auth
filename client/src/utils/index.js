import checkPropTypes from "check-prop-types";
import { applyMiddleware, createStore } from "redux";
import rootReducer from "../reducers";
import { middleware } from "../store";

export const isEmpty = value =>
  value === undefined ||
  value === null ||
  (typeof value === "object" && Object.keys(value).length === 0) ||
  (typeof value === "string" && value.trim().length === 0);

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const checkProps = (component, expectedProps) => {
  return checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );
};

export const isSecure = password => {
  let error_message = "";
  let lowerRegex = /[a-z]/;
  let upperRegex = /[A-Z]/;
  let numberRegex = /[0-9]/;
  let specialRegex = /[$@$!%*?&]/;

  if (isEmpty(password)) {
    error_message = "Password field is required.";
  }
  if (!password.match(lowerRegex))
    error_message = "Password must contain at least one lowercase letter.";
  if (!password.match(upperRegex))
    error_message = "Password must contain at least one uppercase letter.";
  if (!password.match(numberRegex))
    error_message = "Password must contain at least one number.";
  if (!password.match(specialRegex))
    error_message = "Password must contain at least a special character";
  if (password.length < 8 || password.length > 30) {
    error_message = "Password must be between 8 and 30 characters.";
  }
  return error_message;
};

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
