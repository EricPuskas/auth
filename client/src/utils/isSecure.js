import isEmpty from "./isEmpty";

const isSecure = password => {
  let error_message = "";
  let lowerRegex = /[a-z]/;
  let upperRegex = /[A-Z]/;
  let numberRegex = /[0-9]/;
  let specialRegex = /[$@$!%*?&]/;

  if (isEmpty(password)) {
    error_message = "Password field is required.";
  }
  if (!password.match(lowerRegex))
    error_message = "Password must contain at least one lowercase letter.";
  if (!password.match(upperRegex))
    error_message = "Password must contain at least one uppercase letter.";
  if (!password.match(numberRegex))
    error_message = "Password must contain at least one number.";
  if (!password.match(specialRegex))
    error_message = "Password must contain at least a special character";
  if (password.length < 8 || password.length > 30) {
    error_message = "Password must be between 8 and 30 characters.";
  }
  return error_message;
};

export default isSecure;
