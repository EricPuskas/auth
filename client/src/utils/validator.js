import { isSecure } from "./index";

const Validator = {
  login: data => {
    const errors = {};
    if (!data.password) {
      errors.password = "Password field is required.";
    } else {
      errors.password = isSecure(data.password);
    }
    if (!data.email) {
      errors.email = "Email field is required.";
    } else if (!/^[^@]+@[^@]+\.[^@]+$/.test(data.email)) {
      errors.email = "Please enter a valid email address (example@email.com).";
    }
    return errors;
  },
  register: data => {
    const errors = {};
    if (!data.password) {
      errors.password = "Password field is required.";
    } else {
      errors.password = isSecure(data.password);
    }
    if (!data.email) {
      errors.email = "Email field is required.";
    } else if (!/^[^@]+@[^@]+\.[^@]+$/.test(data.email)) {
      errors.email = "Please enter a valid email address (example@email.com).";
    }
    if (!data.firstName) {
      errors.firstName = "First Name field is required.";
    }
    if (!data.lastName) {
      errors.lastName = "Last Name field is required.";
    }
    return errors;
  }
};

export default Validator;
