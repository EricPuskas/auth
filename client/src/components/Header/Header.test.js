import React from "react";
import { shallow } from "enzyme";
import { findByAttribute, testStore } from "../../utils";
import Header from "./Header";

const setup = (initState = {}) => {
  const store = testStore(initState);
  return shallow(<Header store={store} />)
    .childAt(0)
    .dive();
};

describe("Header Component", () => {
  describe("No Auth Test", () => {
    const loggedOutState = {
      auth: {
        clean: false,
        success: false,
        failed: false,
        loading: false,
        user: {
          userId: null,
          firstName: null,
          lastName: null,
          isAuthenticated: false
        }
      }
    };
    let noAuthTest = setup(loggedOutState);
    const test = findByAttribute(noAuthTest, "header-no-auth");
    it("Should correctly display in case the user is not logged in", () => {
      expect(test.length).toBe(1);
    });
    it("Should pass snapshot test", () => {
      expect(noAuthTest.html()).toMatchSnapshot();
    });
  });

  describe("Auth Test", () => {
    const loggedInState = {
      auth: {
        clean: true,
        success: true,
        failed: false,
        loading: false,
        user: {
          userId: 123,
          firstName: "Marko",
          lastName: "Pollo",
          isAuthenticated: true
        }
      }
    };
    let AuthTest = setup(loggedInState);
    const test = findByAttribute(AuthTest, "header-auth");
    it("Should correctly display in case the user is logged in", () => {
      expect(test.length).toBe(1);
    });
    it("Should pass snapshot test", () => {
      expect(AuthTest.html()).toMatchSnapshot();
    });
  });
});
