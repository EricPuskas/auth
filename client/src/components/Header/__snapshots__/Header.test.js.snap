// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Auth Test Should pass snapshot test 1`] = `"<div class=\\"Header\\" data-test=\\"header\\"><h1 data-test=\\"header-auth\\"> Welcome Marko Pollo!</h1></div>"`;

exports[`Header Component No Auth Test Should pass snapshot test 1`] = `"<div class=\\"Header\\" data-test=\\"header\\"><h1 data-test=\\"header-no-auth\\"> Log in to your account </h1></div>"`;
